drivers: sensor: amg88xx: Update driver to use new driver mechanisms

galak opened the pull request:
Move driver to use gpio_dt_spec and DEVICE_DT_GET for handling access
to bus parent device struct.

Signed-off-by: Kumar Gala <galak@kernel.org>

MaureenHelm reviewed (changes requested):
The I2C and GPIO changes should be in separate commits.

Also, why did you rename i2c to i2c_bus_dev? The change adds some unnecessary noise to the diff

galak committed:
drivers: sensor: amg88xx: Update driver to use i2c_dt_spec

Move driver to use i2c_dt_spec for I2C bus access.

Signed-off-by: Kumar Gala <galak@kernel.org>

galak committed:
drivers: sensor: amg88xx: Update driver to use gpio_dt_spec

Move driver to use gpio_dt_spec for GPIO interrupt handling.

Signed-off-by: Kumar Gala <galak@kernel.org>

galak committed:
samples: amg88xx: Convert sample to use DEVICE_DT_GET_ONE

Move to use DEVICE_DT_GET_ONE instead of device_get_binding as we
work on phasing out use of DTS 'label' property.

Signed-off-by: Kumar Gala <galak@kernel.org>

gmarull commented on drivers/sensor/amg88xx/amg88xx_trigger.c:
```diff
@@ -165,24 +163,26 @@ int amg88xx_init_interrupt(const struct device *dev)
-	drv_data->gpio_pin = config->gpio_pin;
+		LOG_DBG("%s: int_gpio not defined in DT", dev->name);
+		return 0;
+	}
```
Driver uses `GPIO_DT_SPEC_INST_GET`, not `GPIO_DT_SPEC_INST_GET_OR`, meaning `int_gpio.port` can't be NULL. We should also return `-ENODEV`.

galak commented on drivers/sensor/amg88xx/amg88xx.h:
```diff
@@ -80,10 +78,10 @@ struct amg88xx_data {
 	const struct device *dev;
-	const struct device *gpio;
-	uint8_t gpio_pin;
+	const struct amg88xx_config *cfg;
```
removed.  getting `config` from `dev` pointer now.

gmarull commented on samples/sensor/amg88xx/src/main.c:
```diff
@@ -51,11 +51,10 @@ void print_buffer(void *ptr, size_t l)
 	int ret;
-	const struct device *dev = device_get_binding(
-				DT_LABEL(DT_INST(0, panasonic_amg88xx)));
+	const struct device *dev = DEVICE_DT_GET(DT_INST(0, panasonic_amg88xx));
```
```suggestion
	const struct device *dev = DEVICE_DT_GET_ONE(panasonic_amg88xx);
```
